src/hiccup/util.clj

```diff
@@ -123,3 +124,16 @@
             (str "?" (url-encode params))
             params)))))
 
+(defn fast-run
+  [callback coll]
+  (when coll
+    (let [^java.util.Iterator iterator (.iterator ^Iterable coll)]
+      (while (.hasNext iterator)
+        (callback (.next iterator))))))
```

Comment:
How does `fast-run` compare with `clojure.core/run!` in terms of performance?

Reply:
Yes, iteration on an iterator is the fastest way to iterate in Clojure

```clj
(def coll (vec (range 100)))

(criterium.core/quick-bench (fast-run inc coll))
;; Evaluation count : 2042790 in 6 samples of 340465 calls.
;;              Execution time mean : 388.494584 ns
;;     Execution time std-deviation : 12.352552 ns
;;    Execution time lower quantile : 375.711953 ns ( 2.5%)
;;    Execution time upper quantile : 403.190971 ns (97.5%)
;;                    Overhead used : 7.814374 ns

(criterium.core/quick-bench (run! inc coll))
;; Evaluation count : 919644 in 6 samples of 153274 calls.
;;              Execution time mean : 681.455647 ns
;;     Execution time std-deviation : 73.381543 ns
;;    Execution time lower quantile : 584.154932 ns ( 2.5%)
;;    Execution time upper quantile : 744.873572 ns (97.5%)
;;                    Overhead used : 7.814374 ns

```
Preferably renamed to **fast-run!**

Reply:
What if it it was called `iterate!`? That would emphasize the iteration part of it.

Reply:
Also, I notice that both `fast-run` and `builld-string` are only used in the compiler namespace. It might be better to move those functions to `hiccup.compiler`, as they're not functions/macros we want to expose as part of the public API.

Reply:
e1776cb882f79d6ca765e9a3d0847cb77a78e72b
2889d7125140f49359efc02608cfd83f74bbf1ef